examples/config-with-defaults/src/test/java/gov/nasa/jpl/aerie/configwithdefaults/ConfigurationTest.java

```diff
@@ -18,18 +18,18 @@
 public final class ConfigurationTest {
 
   @RegisterExtension
-  public static final MerlinExtension<Mission> ext = new MerlinExtension<>();
+  public static final MerlinExtension<ActivityTypes, Mission> ext = new MerlinExtension<>();
 
   private final Mission model;
 
-  public ConfigurationTest(final MerlinTestContext<Mission> ctx)
+  public ConfigurationTest(final MerlinTestContext<ActivityTypes, Mission> ctx)
   throws ConfigurationType.UnconstructableConfigurationException
   {
     // Rely on config. defaults by instantiating config. with empty argument map
     final var config = new ConfigurationMapper().instantiate(Map.of());
 
     this.model = new Mission(ctx.registrar(), config);
```

Comment:
We now have two separate registrars - one for directives, and one for resources/topics. I wonder if it would make sense to either rename the old registrar to make it clearly a separate concept, or toy with the idea of merging the two.

INSTA-EDIT: Thinking about it for 2 seconds I'm realizing that they're called as part of two different workflows - directives are registered by the framework, automatically, while resources and topics are registered by the users. Not sure which argument that supports though...

Reply:
> toy with the idea of merging the two

As you say, they're used at two separate, and independently important, parts of the model instantiation protocol, so I don't think they can be merged. There's probably some renaming that could be done. As noted in the future work section, resources should *also* be registered via the new (or a similarly-timed) registrar; that would make the old registrar not register much of anything, and hence be in dire need of a new name. But until then, some form of registration is happening at two separate times. 🤷‍♂️

Reply:
Also, changing when resources are registered would be a real breaking change for users who depend on being able to register things with reference to a live, constructed model. It would be much more than a refactor!